### ISSUE TITLE
Restock.lic reads incorrect adjective off fest pack resulting in a malformed "rummage" command

### ISSUE THREAD

**Issue by Nyctus** [rpherbig/dr-scripts#4914]
**Summary**:
The item "sleek shadowy black Elothean silk pack stitched with pale ghostflowers" is an unusual festival back-worn pack with a two-word adjective. The pack responds to tap via "tap shadowy pack" and not "tap black pack." Restock.lic appears to be automatically reading "black" as the adjective instead of "shadowy" when it goes to rummage which causes that command to fail.

**Repro**:
1. Log in as an Elothean
2. Acquire a "sleek shadowy black Elothean silk pack stitched with pale ghostflowers" or any of its permutations (e.g. twilight, heretic black, etc)
3. RUB the pack until it is "shadowy black"
4. Set up the YAML to restock "brass razor" as tk_ammo in Shard.
5. Ensure there are fewer brass razors in the pack than is specified in the YAML
6. Run restock.lic

**Results**:
Restock attempts to "rummage" with an invalid adjective.
```
[restock]>tap my brass razor
You tap a steel-bladed brass razor inside your shadowy black pack.
>
[restock]>rummage /C razor IN MY shadowy black pack
I don't know what you are referring to.
```

**Expected**:
Restock uses the correct adjective (in this specific case, "shadowy") or uses the container's noun ("pack") on its own.

Screenshot of my the item in my hands.
![shadowypack](https://user-images.githubusercontent.com/44222707/115491417-cc908100-a214-11eb-881a-7b838d6a2380.png)

**Tap w/ adjective**:
![image](https://user-images.githubusercontent.com/44222707/115492288-61e04500-a216-11eb-8145-27e1404cde49.png)

**Tap w/o adjective**:
![image](https://user-images.githubusercontent.com/44222707/115492382-92c07a00-a216-11eb-836e-e751cbfa840a.png)

**Comment by KatoakDR** [comment]
Thanks for the well documented issue. Will get this fixed soon!

**Comment by KatoakDR** [comment]
Lengthy conversation in Lich discord about how to solve (or not) this issue at https://discord.com/channels/745675889622384681/745676101392793651/834255065111068723

TL;DR
- Simplest approach, don't use odd 3-word containers for items you replenish via `restock`.
- We modify `restock`'s `count_nonstackable_item` method to split the "container" name by spaces and only use the first and last words. If that doesn't match the container then try the second and last words. If that doesn't match then use the last word only.
- We modify `restock` to support a new optional yaml setting, `container:` on the items being restocked so that you can tell the script which container to count and replenish items in. If setting not defined then continue with current behavior of inferring the container.
